core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildLocalContainerRunner.java

```diff
@@ -2,37 +2,112 @@
-            String gid = getLinuxID("-gr");
-            if (uid != null && gid != null && !uid.isEmpty() && !gid.isEmpty()) {
-                Collections.addAll(containerRuntimeArgs, "--user", uid + ":" + gid);
-                if (containerRuntime == ContainerRuntimeUtil.ContainerRuntime.PODMAN) {
-                    // Needed to avoid AccessDeniedExceptions
-                    containerRuntimeArgs.add("--userns=keep-id");
+            if (isDockerRootless(containerRuntime)) {
+                Collections.addAll(containerRuntimeArgs, "--user", String.valueOf(0));
+            } else {
+                String uid = getLinuxID("-ur");
+                String gid = getLinuxID("-gr");
+                if (uid != null && gid != null && !uid.isEmpty() && !gid.isEmpty()) {
+                    Collections.addAll(containerRuntimeArgs, "--user", uid + ":" + gid);
+                    if (containerRuntime == ContainerRuntimeUtil.ContainerRuntime.PODMAN) {
+                        // Needed to avoid AccessDeniedExceptions
+                        containerRuntimeArgs.add("--userns=keep-id");
+                    }
                 }
             }
             baseContainerRuntimeArgs = containerRuntimeArgs.toArray(baseContainerRuntimeArgs);
         }
     }
 
+    private static boolean isDockerRootless(ContainerRuntimeUtil.ContainerRuntime containerRuntime) {
+        if (containerRuntime != ContainerRuntimeUtil.ContainerRuntime.DOCKER) {
+            return false;
+        }
+        String dockerEndpoint = fetchDockerEndpoint();
+        // docker socket?
+        String socketUriPrefix = "unix://";
+        if (dockerEndpoint == null || !dockerEndpoint.startsWith(socketUriPrefix)) {
+            return false;
+        }
+        String dockerSocket = dockerEndpoint.substring(socketUriPrefix.length());
+        String currentUid = getLinuxID("-ur");
+        if (currentUid == null || currentUid.isEmpty() || currentUid.equals(String.valueOf(0))) {
+            return false;
+        }
+
+        int socketOwnerUid;
+        try {
+            socketOwnerUid = (int) Files.getAttribute(Path.of(dockerSocket), "unix:uid", LinkOption.NOFOLLOW_LINKS);
+        } catch (IOException e) {
+            LOGGER.infof("Owner UID lookup on '%s' failed with '%s'", dockerSocket, e.getMessage());
+            return false;
+        }
+        return currentUid.equals(String.valueOf(socketOwnerUid));
+    }
+
+    private static String fetchDockerEndpoint() {
+        // DOCKER_HOST environment variable overrides the active context
+        String dockerHost = System.getenv("DOCKER_HOST");
+        if (dockerHost != null) {
+            return dockerHost;
+        }
+
+        OutputFilter outputFilter = new OutputFilter();
+        if (!ExecUtil.execWithTimeout(new File("."), outputFilter, Duration.ofMillis(3000),
+                "docker", "context", "ls", "--format",
+                "'{{- if .Current -}} {{- .DockerEndpoint -}} {{- end -}}'")) {
```

Comment:
```suggestion
                "{{- if .Current -}} {{- .DockerEndpoint -}} {{- end -}}")) {
```

With the `'` it returns `'''unix:///run/user/1000/docker.sock'` which fails the `startsWith("unix://")` check

Reply:
It's essentially like running:
```
$ docker context ls --format "'{{- if .Current -}} {{ .DockerEndpoint -}} {{- end -}}'"
''
'unix:///run/user/1000/docker.sock'
```

Reply:
Good catch